docker/utils/utils.py

```diff
@@ -518,6 +518,27 @@ def create_host_config(
     return host_config
 
 
+def parse_env_file(env_file):
+    """
+    Reads a line-separated environment file.
+    The format of each line should be "key=value".
+    """
+    environment = {}
+
+    with open(env_file, 'r') as f:
+        for line in f:
+            parse_line = line.strip().split('=')
+            if len(parse_line) == 2 and line[0] != '#':
+                k = parse_line[0]
+                v = parse_line[1]
+                environment[k] = v
+            elif line[0] != '#':
+                raise errors.DockerException(
+                    'Invalid env_file line:\n{0}'.format(line))
```

Comment:
Rather than checking twice for `line[0] != '#'`, I think this could be simplified to:

```python
if line[0] == '#':
    continue

parse_line = line.strip().split('=')
if len(parse_line) == 2:
    # ...
else:
    # ...
```